### PR TITLE
upgrade flow bin to ^0.89.0

### PR DESCRIPTION
**Pull request by esetnik**

### CODE REVIEW

**Comment by erikras** [comment]
I've been wanting to upgrade flow for a while now, but the later versions of flow are incompatible with some of the third party libraries, like immutable and lodash, and there's just no way to get the project to validate. Or at least that I've been able to accomplish. I'd love for someone to figure out how.

**Comment by esetnik** [comment]
Unfortunately i'm probably not the right person to fix this as I use typescript not flow. Would you like me to close this PR?

File: package.json [hunk 1]
```diff
@@ -110,7 +110,7 @@
     "eslint-plugin-import": "^2.14.0",
     "eslint-plugin-jsx-a11y": "^6.1.2",
     "eslint-plugin-react": "^7.11.1",
-    "flow-bin": "^0.72.0",
+    "flow-bin": "^0.89.0",
```
Comment (iamandrewluca) [inline package.json]: let's first update to latest `^0.79.1` 
```suggestion
    "flow-bin": "^0.79.1",
```

```basg
⇒  npm run test:flow      

> redux-form@8.2.4 test:flow /home/aluca/Projects/redux-form
> node scripts/patch-immutable-flow-error.js && opencollective-postinstall && flow check

Thank you for using redux-form!
If you rely on this package, please consider supporting our open collective:
> https://opencollective.com/redux-form/donate

Found 0 errors
```

Comment (iamandrewluca) [reply package.json]: also please rebase

**Pull request closed without merging**